kprobe_queue.c

```diff
@@ -274,28 +284,55 @@ struct quark_queue_ops queue_ops_kprobe = {
 	.close	      = kprobe_queue_close,
 };
 
+static ssize_t	parse_data_offset(const char *);
+
 static char *
 str_of_dataloc(struct perf_record_sample *sample,
     struct perf_sample_data_loc *data_loc)
 {
 	return (sample->data + data_loc->offset);
 }
 
-static inline int
-sample_kind_of_id(struct kprobe_queue *kqq, int id)
+static struct sample_attr *
```

Comment:
In this specific case I can't because it's returned on``sample_attr_prepare`, so I'd have to deconstify.
But I took your suggestion and constified the other cases here: https://github.com/elastic/quark/pull/94/commits/6912f150ee5c6be0acbb4119d3971286c7736c54